src/Calypso-SystemPlugins-SUnit-Browser/ClyJumpToTestMethodCommand.class.st

```diff
@@ -0,0 +1,98 @@
+"
+Description
+--------------------
+
+I am a command available in the method context menu of Calypso. When activated I'll jump on the test method corresponding to the method. 
+
+If the method does not exists I will create it in the corresponding class.
+
+Internal Representation and Key Implementation Points.
+--------------------
+
+    Instance Variables
+	selectedMethods:			<aCollection>	Methods selected by the user that should have their tests generated.
+	testMethodToBrowse:		<aMethod>			The method to browse at the end of the execution of the command.
+
+"
+Class {
+	#name : #ClyJumpToTestMethodCommand,
+	#superclass : #CmdCommand,
+	#traits : 'TClyGenerateTestClass',
+	#classTraits : 'TClyGenerateTestClass classTrait',
+	#instVars : [
+		'selectedMethods',
+		'testMethodToBrowse'
+	],
+	#category : #'Calypso-SystemPlugins-SUnit-Browser-Commands'
+}
+
+{ #category : #testing }
+ClyJumpToTestMethodCommand class >> canBeExecutedInContext: aBrowserContext [
+	^ aBrowserContext isInstanceSideMethodSelected and: [ aBrowserContext selectedMethods anySatisfy: [ :each | each isTestMethod not ] ]
+]
+
+{ #category : #activation }
+ClyJumpToTestMethodCommand class >> methodMenuActivation [
+	<classAnnotation>
+	^ CmdContextMenuActivation byItemOf: ClySUnitMethodMenuGroup order: 10 for: ClyMethod asCalypsoItemContext
+]
+
+{ #category : #activation }
+ClyJumpToTestMethodCommand class >> methodShortcutActivation [
+	<classAnnotation>
+	^ CmdShortcutActivation by: $h meta , $j meta for: ClyMethod asCalypsoItemContext
+]
+
+{ #category : #execution }
+ClyJumpToTestMethodCommand >> applyResultInContext: aToolContext [
+	super applyResultInContext: aToolContext.
+	testMethodToBrowse ifNotNil: [ :testMethod | aToolContext showMethod: testMethod ]
+]
+
+{ #category : #accessing }
+ClyJumpToTestMethodCommand >> defaultMenuItemName [
+	^ 'Jump to test method'
+]
+
+{ #category : #execution }
+ClyJumpToTestMethodCommand >> execute [
+	selectedMethods ifEmpty: [ self inform: 'No method selected'. ^ self ].
+	
+	selectedMethods do: [ :method | self generateTestMethodFor: method ]
```

Comment:
Same comment as for jump to class. If #execute performs only generation then command should be named accordingly.